### PR TITLE
Prefer object initializers

### PR DESCRIPTION
**Pull request by cadatoiva**
This PR implements the Code Style policy of preferring object initializers. Had to go by hand and move some lines to allow for full initialization instead of partial. Some of them still have 1 or 2 Properties that aren't initialized, but that's due to the complexity of moving the lines out of the way.

Added some TODOs where I found some easy refactoring spots to work with as well.

### CODE REVIEW

**Comment by ngosang** [comment]
@cadatoiva This will need a rebase after we finish #7292 I will review then.

**Comment by cadatoiva** [comment]
I agree

**Comment by cadatoiva** [comment]
This should be ready for review once the checks pass

File: src/Jackett.Common/Indexers/GazelleGames.cs [hunk 1]
```diff
@@ -268,69 +268,48 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-                        }
-                        else
-                        {
-                            release.UploadVolumeFactor = 1;
-                            release.DownloadVolumeFactor = 1;
-                        }
-
-                        releases.Add(release);
+                            MinimumRatio = 1,
+                            MinimumSeedTime = 80 * 3600,
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/GazelleGames.cs]: replace with the multiplication with the result and let a comment with the time in hours

File: src/Jackett.Common/Indexers/HDBitsApi.cs [hunk 2]
```diff
@@ -123,14 +123,29 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
             var releases = new List<ReleaseInfo>();
-
             foreach (JObject r in response["data"])
             {
-                var release = new ReleaseInfo();
-                release.Title = (string)r["name"];
-                release.Comments = new Uri(SiteLink + "details.php?id=" + (string)r["id"]);
-                release.Link = new Uri(SiteLink + "download.php/" + (string)r["filename"] + "?id=" + (string)r["id"] + "&passkey=" + configData.Passkey.Value);
-                release.Guid = release.Link;
+                var guid = new Uri(
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/HDBitsApi.cs]: rename to link

File: src/Jackett.Common/Indexers/HDBitsApi.cs [hunk 3]
```diff
@@ -123,14 +123,29 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                {
+                    Title = (string)r["name"],
+                    Comments = new Uri(SiteLink + "details.php?id=" + (string)r["id"]),
+                    Link = guid,
+                    Category = MapTrackerCatToNewznab((string)r["type_category"]),
+                    Size = (long)r["size"],
+                    Files = (long)r["numfiles"],
+                    Grabs = (long)r["times_completed"],
+                    Seeders = seeders,
+                    PublishDate = DateTimeUtil.UnixTimestampToDateTime((int)r["utadded"]),
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/HDBitsApi.cs]: As I said in other comment, make the conversions that can fail outside the initializer.

File: src/Jackett.Common/Indexers/HDOlimpo.cs [hunk 4]
```diff
@@ -93,20 +94,17 @@ private async Task DoLogin()
...
-            var boundary = "---------------------------" + (DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1))).TotalSeconds.ToString(CultureInfo.InvariantCulture).Replace(".", "");
+            var pairs = new Dictionary<string, string>
+                {
+                    {"freetorrent", "false"},
+                    {"ordenar_por", "created_at"},
+                    {"orden", "desc"},
+                    {"titulo", query.GetQueryString()},
+                    {"categoria", cats.Count == 1 ? cats.First() : "0"}
+                };
+            var boundary = "---------------------------" + DateTimeUtil.DateTimeToUnixTimestamp(DateTime.UtcNow).ToString();
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/HDOlimpo.cs]: I'm not sure about this change, the toString and Replace part. I copied this code from other part of the Jackett code and I know it's working. I prefer to do it in other PR if you want.

File: src/Jackett.Common/Indexers/HDOlimpo.cs [hunk 5]
```diff
@@ -159,49 +157,46 @@ private List<ReleaseInfo> ParseResponse(WebClientStringResult response, bool inc
...
-                    releases.Add(release);
+                    releases.Add(new ReleaseInfo
+                    {
+                        Title = title,
+                        Category = MapTrackerCatToNewznab((string)torrent["categoria"]),
+                        Size = (long)torrent["size"],
+                        Grabs = (long)torrent["snatched"],
+                        InfoHash = (string)torrent["plain_info_hash"],
+                        Link = new Uri(DownloadUrl + (string)torrent["id"]),
+                        Files = ((JArray)JsonConvert.DeserializeObject<dynamic>((string)torrent["files_list"])).Count,
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/HDOlimpo.cs]: As I said in other comment, make the conversions that can fail outside the initializer.

File: src/Jackett.Common/Indexers/LostFilm.cs [hunk 6]
```diff
@@ -707,27 +712,35 @@ private async Task<List<ReleaseInfo>> FollowTrackerRedirection(string url, Track
...
-                        release.Size = ReleaseInfo.GetBytes(sizeString);
-
-                        // add missing torznab fields not available from results
-                        release.Seeders = 1;
-                        release.Peers = 2;
-                        release.DownloadVolumeFactor = 0;
-                        release.UploadVolumeFactor = 1;
-                        release.MinimumRatio = 1;
-                        release.MinimumSeedTime = 172800; // 48 hours
+                        var guid = new Uri(downloadLink.GetAttribute("href"));
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/LostFilm.cs]: The correct var name is link or downloadLink. guid means unique id and can be anything, in other trackers we use the comment link, the magnet link.

File: src/Jackett.Common/Indexers/HDBitsApi.cs (hunk 3 above)
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/HDBitsApi.cs]: Does that include all the `(string)` and `(long)` in this one?

File: src/Jackett.Common/Indexers/BroadcastTheNet.cs [hunk 7]
```diff
@@ -160,7 +139,36 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                            Grabs = btnResult.Snatched,
+                            Description = string.Join("<br />\n", descriptions)
+                        };
+
+                        if (!string.IsNullOrEmpty(btnResult.SeriesBanner))
+                            item.BannerUrl = new Uri(btnResult.SeriesBanner);
+                        if (item.Category.Count == 0) // default to TV
+                            item.Category.Add(TorznabCatType.TV.ID);
+                        if (!string.IsNullOrWhiteSpace(btnResult.ImdbID))
+                            item.Imdb = ParseUtil.CoerceLong(btnResult.ImdbID);
```
[the remark this answers is not in the document]
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/BroadcastTheNet.cs]: Changed

File: src/Jackett.Common/Indexers/HDBitsApi.cs (hunk 3 above)
Comment (ngosang) [reply src/Jackett.Common/Indexers/HDBitsApi.cs]: Keep casting inside the initializer is ok for me.

File: src/Jackett.Common/Indexers/Torrentscsv.cs [hunk 8]
```diff
@@ -97,11 +98,8 @@ public async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query, int
                     if (torrent == null)
                         throw new Exception("Error: No data returned!");
 
-                    var release = new ReleaseInfo();
-                    release.Title = torrent.Value<string>("name");
-
                     // construct magnet link from infohash with all public trackers known to man
-                    var magnet_uri = "magnet:?xt=urn:btih:" + torrent.Value<JToken>("infohash") +
+                    var magnet_uri = new Uri( "magnet:?xt=urn:btih:" + torrent.Value<JToken>("infohash") +
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/Torrentscsv.cs]: Add a TODO, we should have a global var with public bittorent trackers so we can reuse it in all trackers.

File: src/Jackett.Common/Indexers/Torrentscsv.cs [hunk 9]
```diff
@@ -134,32 +132,35 @@ public async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query, int
...
+                    {
+                        Title = torrent.Value<string>("name"),
+                        MagnetUri = magnet_uri,
+                        // there is no comments or details link so we point to the web site instead
+                        Comments = new Uri(SiteLink),
+                        Guid = magnet_uri,
+                        Link = magnet_uri,
+                        InfoHash = torrent.Value<JToken>("infohash").ToString(),
+                        PublishDate = dateTime,
+                        Seeders = releaseSeeders,
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/Torrentscsv.cs]: seeders

File: src/Jackett.Common/Indexers/Torrentscsv.cs [hunk 10]
```diff
@@ -134,32 +132,35 @@ public async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query, int
...
-
+                    var releaseSeeders = torrent.Value<int>("seeders");
+                    var grabs = ParseUtil.CoerceInt(torrent.Value<string>("completed") ?? "0");
+                    var release = new ReleaseInfo
+                    {
+                        Title = torrent.Value<string>("name"),
+                        MagnetUri = magnet_uri,
+                        // there is no comments or details link so we point to the web site instead
+                        Comments = new Uri(SiteLink),
+                        Guid = magnet_uri,
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/Torrentscsv.cs]: magnet (you don't need to say it's a Uri, it's implicit)

File: src/Jackett.Common/Indexers/Xthor.cs [hunk 11]
```diff
@@ -234,6 +234,10 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                             torrent.name = regex.Replace(torrent.name, "$1" + ReplaceMulti + "$2");
                         }
 
+                        var publishDate = DateTimeUtil.UnixTimestampToDateTime(torrent.added);
+                        var guid = new Uri(TorrentDescriptionUrl.Replace("{id}", torrent.id.ToString()));
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/Xthor.cs]: I think you can remove all the code related to guid and use the link instead, is unique.

File: src/Jackett.Common/Indexers/ShowRSS.cs [hunk 12]
```diff
@@ -67,42 +67,39 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-                    release.Seeders = 1;
-                    release.Peers = 1;
-                    release.DownloadVolumeFactor = 0;
-                    release.UploadVolumeFactor = 1;
-                    release.MagnetUri = new Uri(node.SelectSingleNode("link").InnerText);
-                    releases.Add(release);
+                    var magnetUri = new Uri(node.SelectSingleNode("link").InnerText);
+                    var publishDate = DateTime.Parse(node.SelectSingleNode("pubDate").InnerText, CultureInfo.InvariantCulture);
+                    var infoHash = node.SelectSingleNode("description").InnerText;
+                    var guid = new Uri(test.Attributes["url"].Value);
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/ShowRSS.cs]: I think you can remove all the code related to guid and use the magnet instead

File: src/Jackett.Common/Indexers/pornolab.cs [hunk 13]
```diff
@@ -278,51 +279,46 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
                         {
-                            var regex = new Regex(@"(\([А-Яа-яЁё\W]+\))|(^[А-Яа-яЁё\W\d]+\/ )|([а-яА-ЯЁё \-]+,+)|([а-яА-ЯЁё]+)");
-                            release.Title = regex.Replace(release.Title, "");
-                        }
-
-                        releases.Add(release);
+                            MinimumRatio = 1,
+                            MinimumSeedTime = 0,
+                            Title =
+                                title,
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/pornolab.cs]: indent

File: src/Jackett.Common/Indexers/pornolab.cs [hunk 14]
```diff
@@ -278,51 +279,46 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-
-                        if (configData.StripRussianLetters.Value)
+                        var title = configData.StripRussianLetters.Value
+                            ? s_StripRussianRegex.Replace(qDetailsLink.TextContent, "")
+                            : qDetailsLink.TextContent;
+                        var size = ReleaseInfo.GetBytes(qSize.TextContent);
+                        var leechers = ParseUtil.CoerceInt(Row.QuerySelector("td:nth-child(8)").TextContent);
+                        var grabs = ParseUtil.CoerceLong(Row.QuerySelector("td:nth-child(9)").TextContent);
+                        var publishDate = DateTimeUtil.UnixTimestampToDateTime(long.Parse(timestr));
+                        releases.Add(new ReleaseInfo
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/pornolab.cs]: don't do this please 😢

File: src/Jackett.Common/Indexers/rutracker.cs [hunk 15]
```diff
@@ -1539,42 +1539,46 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                        var size = ReleaseInfo.GetBytes(qSize.GetAttribute("data-ts_text"));
+                        var leechers = ParseUtil.CoerceInt(Row.QuerySelector("td:nth-child(8)").TextContent);
+                        var grabs = ParseUtil.CoerceLong(Row.QuerySelector("td:nth-child(9)").TextContent);
+                        var publishDate = DateTimeUtil.UnixTimestampToDateTime(long.Parse(timestr));
+                        var release = new ReleaseInfo
+                        {
+                            MinimumRatio = 1,
+                            MinimumSeedTime = 0,
+                            Title = qDetailsLink.TextContent,
+                            Comments = releaseComments,
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/rutracker.cs]: rename comments

File: src/Jackett.Common/Indexers/toloka.cs [hunk 16]
```diff
@@ -247,48 +247,48 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                        var leechers = ParseUtil.CoerceInt(Row.QuerySelector("td:nth-child(11) > b").TextContent);
+                        var publishDate = DateTimeUtil.FromFuzzyTime(timestr);
+                        var release = new ReleaseInfo
+                        {
+                            MinimumRatio = 1,
+                            MinimumSeedTime = 0,
+                            Title = qDetailsLink.TextContent,
+                            Comments = releaseComments,
+                            Link = link,
+                            Guid = releaseComments,
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/toloka.cs]: rename comments

File: src/Jackett.Common/Indexers/toloka.cs [hunk 17]
```diff
@@ -247,48 +247,48 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+
+                        // TODO cleanup
                         if (release.Category.Contains(TorznabCatType.TV.ID))
                         {
                             // extract season and episodes
                             var regex = new Regex(".+\\/\\s([^а-яА-я\\/]+)\\s\\/.+Сезон\\s*[:]*\\s+(\\d+).+(?:Серії|Епізод)+\\s*[:]*\\s+(\\d+-*\\d*).+,\\s+(.+)\\]\\s(.+)");
 
-                            var title = regex.Replace(release.Title, "$1 - S$2E$3 - rus $4 $5");
+                            string title;
+                            title = regex.Replace(release.Title, "$1 - S$2E$3 - rus $4 $5");
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/toloka.cs]: var title =  ??

File: src/Jackett.Common/Services/CacheService.cs [hunk 18]
```diff
@@ -19,24 +19,17 @@ public void CacheRssResults(IIndexer indexer, IEnumerable<ReleaseInfo> releases)
         {
             lock (cache)
             {
-                var trackerCache = cache.Where(c => c.TrackerId == indexer.ID).FirstOrDefault();
-                if (trackerCache == null)
-                {
-                    trackerCache = new TrackerCache();
-                    trackerCache.TrackerId = indexer.ID;
-                    trackerCache.TrackerName = indexer.DisplayName;
+                var trackerCache = cache.FirstOrDefault(c => c.TrackerId == indexer.ID) ??
```
Comment (ngosang) [inline src/Jackett.Common/Services/CacheService.cs]: I think this logic is not the same, you are calling `cache.Add(trackerCache);` in all cases.
The changes in this class are not related to this PR, please undo.

File: src/Jackett.Common/Indexers/ShowRSS.cs (hunk 12 above)
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/ShowRSS.cs]: Not sure I want to do that without testing it first to see why this particular value was used. I'll add TODO comment for future maintenance for now to bring attention to it.

File: src/Jackett.Common/Indexers/TorrentHeaven.cs [hunk 19]
```diff
@@ -192,50 +193,50 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-                    var dateGerman = dateStrParts[0] switch
-                    {
-                        "Heute" => (DateTime.SpecifyKind(DateTime.UtcNow.Date, DateTimeKind.Unspecified) + TimeSpan.Parse(dateStrParts[1])),
-                        "Gestern" => (DateTime.SpecifyKind(DateTime.UtcNow.Date, DateTimeKind.Unspecified) + TimeSpan.Parse(dateStrParts[1])
-                                      - TimeSpan.FromDays(1)),
-                        _ => DateTime.SpecifyKind(                            DateTime.ParseExact(
-                                dateStrParts[0] + dateStrParts[1], "dd.MM.yyyyHH:mm", CultureInfo.InvariantCulture),
-                            DateTimeKind.Unspecified)
-                    };
+                    var releaseSeeders = ParseUtil.CoerceInt(qSeeders.TextContent);
```
[the remark this answers is not in the document]
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/TorrentHeaven.cs]: I tried to reduce diff size because I knew how large this kind of refactor was going to be. So this was affected by the order I was making changes to the code, specifically going ahead of your initial review to pull out other values inside the initializer.

File: src/Jackett.Common/Indexers/Xthor.cs (hunk 11 above)
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/Xthor.cs]: I'll add a TODO to this one as well, as I don't want to change something until I know why it is the way it is.

File: src/Jackett.Common/Services/CacheService.cs (hunk 18 above)
Comment (cadatoiva) [reply src/Jackett.Common/Services/CacheService.cs]: Yeah, I see that now. I was attempting to refactor since it was already going to change from the initializer.

File: src/Jackett.Common/Indexers/Torrentscsv.cs (hunk 8 above)
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/Torrentscsv.cs]: In all **public** trackers. Private trackers don't like when you share their stuff externally like that.

File: src/Jackett.Common/Indexers/720pier.cs [hunk 20]
```diff
@@ -186,55 +186,51 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                                            .Replace("KiB", "KB")
+                                            .Replace("ГБ", "GB")
+                                            .Replace("МБ", "MB")
+                                            .Replace("КБ", "KB");
+                        var comments = new Uri(SiteLink + qDetailsLink.GetAttribute("href"));
+                        var leechers = ParseUtil.CoerceInt(row.QuerySelector("span.leech").TextContent);
+                        var link = new Uri(SiteLink + qDownloadLink.GetAttribute("href").TrimStart('/'));
+                        var publishDate = DateTimeUtil.FromUnknown(timestr, "UK");
+                        var size = ReleaseInfo.GetBytes(sizeString);
+                        releases.Add(new ReleaseInfo
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/720pier.cs]: var release = ...

File: src/Jackett.Common/Indexers/GazelleGames.cs [hunk 21]
```diff
@@ -268,68 +269,49 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-                        {
-                            release.UploadVolumeFactor = 1;
-                            release.DownloadVolumeFactor = 1;
-                        }
-
+                            MinimumRatio = 1,
+                            MinimumSeedTime = 288000, //80 hours
+                            Category = GroupCategory,
+                            PublishDate = publishDate,
+                            Size = ReleaseInfo.GetBytes(size),
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/GazelleGames.cs]: move outside

File: src/Jackett.Common/Indexers/NewRealWorld.cs [hunk 22]
```diff
@@ -168,15 +164,10 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-                    var release = new ReleaseInfo();
-                    release.MinimumRatio = 0.75;
-                    release.MinimumSeedTime = 0;
-
 
                     var qDetailsLink = row.QuerySelector("a[href^=details.php?id=]");
-                    release.Title = qDetailsLink.Text();
 
-                    if (!query.MatchQueryStringAND(release.Title))
+                    if (!query.MatchQueryStringAND(qDetailsLink.Text()))
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/NewRealWorld.cs]: we are using this code below, just declare `var title = qDetailsLink.TextContent`

File: src/Jackett.Common/Indexers/Pretome.cs [hunk 23]
```diff
@@ -284,46 +284,44 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-                    release.DownloadVolumeFactor = 0; // ratioless
-                    release.UploadVolumeFactor = 1;
-
-                    releases.Add(release);
+                    var link = new Uri(SiteLink + qDownload.GetAttribute("href"));
+                    var publishDate = DateTimeUtil.FromTimeAgo(dateStr);
+                    var size = ReleaseInfo.GetBytes(sizeStr);
+                    var leechers = ParseUtil.CoerceInt(row.Children[10].TextContent);
+                    var grabs = ParseUtil.CoerceInt(row.QuerySelector("td:nth-child(9)").TextContent);
+                    releases.Add(new ReleaseInfo
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/Pretome.cs]: var release = ...

File: src/Jackett.Common/Indexers/Rarbg.cs [hunk 24]
```diff
@@ -235,41 +239,49 @@ private async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query, in
...
+                    {
+                        Title = title,
+                        Category = MapTrackerCatDescToNewznab(item.Value<string>("category")),
+                        MagnetUri = magnetUri,
+                        InfoHash = infoHash,
+                        Comments = comments,
+                        Link = link,
+                        PublishDate = publishDate,
+                        Guid = magnetUri,
+                        Seeders = releaseSeeders,
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/Rarbg.cs]: seeders

File: src/Jackett.Common/Indexers/ShowRSS.cs [hunk 25]
```diff
@@ -67,42 +67,40 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
+                    var publishDate = DateTime.Parse(node.SelectSingleNode("pubDate").InnerText, CultureInfo.InvariantCulture);
+                    var infoHash = node.SelectSingleNode("description").InnerText;
+                    //TODO Maybe use magnetUri instead? https://github.com/Jackett/Jackett/pull/7342#discussion_r397552678
+                    var guid = new Uri(test.Attributes["url"].Value);
+                    releases.Add(new ReleaseInfo
+                    {
+                        MinimumRatio = 1,
+                        MinimumSeedTime = 172800, // 48 hours
+                        Title = title,
+                        Comments = magnetUri,
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/ShowRSS.cs]: you made a mistake here

File: src/Jackett.Common/Services/CacheService.cs [hunk 26]
```diff
@@ -19,26 +19,23 @@ public void CacheRssResults(IIndexer indexer, IEnumerable<ReleaseInfo> releases)
...
                 }
 
                 foreach (var release in releases.OrderByDescending(i => i.PublishDate))
                 {
-                    var existingItem = trackerCache.Results.Where(i => i.Result.Guid == release.Guid).FirstOrDefault();
-                    if (existingItem == null)
-                    {
-                        existingItem = new CachedResult();
-                        existingItem.Created = DateTime.Now;
+                    var existingItem = trackerCache.Results.FirstOrDefault(i => i.Result.Guid == release.Guid) ??
```
Comment (ngosang) [inline src/Jackett.Common/Services/CacheService.cs]: the same as before, this code is not equivalent, you are calling `trackerCache.Results.Add(existingItem);` always...
please pay more attention. I can't walk behind you forever

File: src/Jackett.Common/Indexers/ShowRSS.cs (hunk 25 above)
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/ShowRSS.cs]: No, compare to Line 86 in original code:
`release.Comments = new Uri(node.SelectSingleNode("link").InnerText);`
and Line 104 of original code:
`release.MagnetUri = new Uri(node.SelectSingleNode("link").InnerText)`

File: src/Jackett.Common/Services/CacheService.cs (hunk 26 above)
Comment (cadatoiva) [reply src/Jackett.Common/Services/CacheService.cs]: In large, sweeping, mostly automated refactoring like this, there are going to be things that fall through the cracks. I wouldn't be doing something like this alone because no matter how good someone is, or how good their tools, there will be mistakes. I do generally pay attention to what I'm doing, and this PR took me like 5 days just to review for the initial submission.

I do what I can, but I'm not perfect.

File: src/Jackett.Common/Indexers/ShowRSS.cs (hunk 25 above)
Comment (ngosang) [reply src/Jackett.Common/Indexers/ShowRSS.cs]: Sorry, not your fault. Just let a comment to fix Comments and Description too. #7846

File: src/Jackett.Common/Indexers/TVstore.cs [hunk 27]
```diff
@@ -224,24 +224,19 @@ protected async Task<bool> GetSeriesInfo()
...
-                        }
+                        var id = seriesKnowBySite[i];
+                        var seriesElement = WebUtility.HtmlDecode(id).Split(';');
+                        series.Add(
+                            new SeriesDetail
+                            {
+                                HunName = seriesElement[1].Split('=')[1].Trim('\'').ToLower(),
+                                EngName = seriesElement[2].Split('=')[1].Trim('\'').ToLower(),
+                                id = seriesElement[0].Split('=')[1].Trim('\''),
+                                imdbid = seriesElement[7].Split('=')[1].Trim('\'')
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/TVstore.cs]: well, we have the parsing problem here too, we won't know what field failed

File: src/Jackett.Common/Indexers/HDOlimpo.cs (hunk 4 above)
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/HDOlimpo.cs]: I know this is already resolved, but due to a conflict in rebasing after the extensions this part came to my attention again. According to https://stackoverflow.com/questions/2305218/what-is-the-boundary-parameter-in-an-http-multi-part-post-request the boundary string doesn't have to be anything in particular except for the dashes. The remainder is a random string that just has to be the same in all places for a specific request.

Comment (ngosang) [reply src/Jackett.Common/Indexers/HDOlimpo.cs]: As I said, just copied and tested => https://github.com/Jackett/Jackett/search?q=%28DateTime.UtcNow.Subtract%28new+DateTime%281970%2C+1%2C+1%29%29&unscoped_q=%28DateTime.UtcNow.Subtract%28new+DateTime%281970%2C+1%2C+1%29%29

File: src/Jackett.Common/Indexers/TVstore.cs [hunk 28]
```diff
@@ -374,7 +366,13 @@ public class SeriesDetail
         public string HunName;
         public string EngName;
         public string imdbid;
-
+        public SeriesDetail(string[] seriesElement)
+        {
+            HunName = seriesElement[1].Split('=')[1].Trim('\'').ToLower();
+            EngName = seriesElement[2].Split('=')[1].Trim('\'').ToLower();
+            id = seriesElement[0].Split('=')[1].Trim('\'');
+            imdbid = seriesElement[7].Split('=')[1].Trim('\'');
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/TVstore.cs]: > well, we have the parsing problem here too, we won't know what field failed

You are not solving the problem...

File: src/Jackett.Common/Indexers/NewRealWorld.cs [hunk 29]
```diff
@@ -181,36 +177,40 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
...
-                    release.DownloadVolumeFactor = row.QuerySelector("img[title=\"OnlyUpload\"]") != null ? 0 : 1;
-                    release.UploadVolumeFactor = 1;
-
-                    releases.Add(release);
+                    var files = ParseUtil.CoerceInt(row.QuerySelector("td:contains(Datei) > strong ~ strong").TextContent);
+                    var comments = new Uri(SiteLink + qDetailsLink.GetAttribute("href"));
+                    var leechers = ParseUtil.CoerceInt(qLeechers.Text());
+                    var size = ReleaseInfo.GetBytes(qSize.TextContent.Replace(".", "").Replace(",", "."));
+                    var downloadVolumeFactor = row.QuerySelector("img[title=\"OnlyUpload\"]") != null ? 0 : 1;
+                    releases.Add(new ReleaseInfo
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/NewRealWorld.cs]: var release = ...

File: src/Jackett.Common/Indexers/TVstore.cs (hunk 28 above)
Comment (cadatoiva) [reply src/Jackett.Common/Indexers/TVstore.cs]: In the constructor a parse error will give a line number, so we will know what field failed. I'm planning on coming back to this section and doing a more thorough refactoring later (with all the TODOs)

Comment (ngosang) [reply src/Jackett.Common/Indexers/TVstore.cs]: Well I remember testing this in the past, if you change this line to (wrong index) `imdbid = seriesElement[7].Split('=')[999].Trim('\'');` I'm pretty sure you won't see the error in line 374.

Please, don't do more changes in this PR, just fix remaining issues and merge.
